api/v1alpha1/ratelimit_types.go

```diff
@@ -42,17 +54,45 @@ type GlobalRateLimit struct {
 	Rules []RateLimitRule `json:"rules"`
 }
 
+// LocalRateLimit defines local rate limit configuration.
+type LocalRateLimit struct {
+
+	// Envoy requires a default rate limit to be set for each route.
+	// The other possible option is to set the default rate limit to unit32 max and set the
+	// fill interval to a short period, like 1 second. This will effectively make
+	// the default rate limit "unlimited".
+	// See https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/local_rate_limit_filter#using-rate-limit-descriptors-for-local-rate-limiting
+
+	// Limit is the default rate limit for a route if no rules match.
+	// If a request does not match any of the rules, it is counted towards this limit.
+	//
+	// Note: Limit is applied per route. Even if a policy targets a gateway,
+	// each route in that gateway still has a separate rate limit bucket.
+	// For example, if a gateway has 2 routes, and the limit is 100r/s, then
+	// each route has its own 100r/s rate limit bucket.
+	Limit RateLimitValue `json:"default"`
```

Comment:
**Why do we need this?** 
Envoy requires a rate limit bucket to be set for each route. If a request does not match any of the rules, it is counted towards the default limit. It's a two-level rate limit mechanism, a default bucket for the whole route, and separate buckets can be defined if more fine-grained rate limits are needed for specific headers or source IP ranges.

The other option is to set the default rate limit to unit32 max and set the fill interval to a very short period, like 1 second. This will effectively make the default rate limit "unlimited".

I'm inclined to add a `Default` bucket to both the Local and Global rate limits. The reasoning is that if the length of `ClientSelectors` of a `RateLimitRule` is 0, that `RateLimitRule` is already effectively a default rate limit for that route, but it's implicit and confusing to users, and what makes this even worse is that descriptors are processed sequentially in Envoy, so if this rule is not the first one in the rules, it won't take effect. Instead, adding a `Default` rate limit explicitly at the top level and requiring the length of `ClientSelectors` at least 1 results in a much more understandable and user-friendly UI.

If a user doesn't want a `Default` rate limit at the route level, we can make it optional and fall back to the unit32 max approach, but I prefer that the `Default` should be mandatory,  otherwise, the behavior of rate limit is undefined when the length of rules is 0.

See https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/local_rate_limit_filter#using-rate-limit-descriptors-for-local-rate-limiting

Need your input here: @arkodg @zirain @envoyproxy/gateway-maintainers

Reply:
https://gateway.envoyproxy.io/v0.6.0/user/rate-limit/#rate-limit-all-requests highlights how you can have a default ratelimit for all requests, but also a specific rate limit for specific clients

Reply:
we may need something like `The other option is to set an implicit default rate limit as unit32 max and set the fill interval to a very short period, like 1 second. This will effectively make the default rate limit "unlimited".` if a default limit value is required, but we must ensure that it is truly `unlimited`

Reply:
> https://gateway.envoyproxy.io/v0.6.0/user/rate-limit/#rate-limit-all-requests

Yes, the current approach also works, but I prefer an explicit "global" setting because

* Explict approach is confusing to the users.
* Descriptors are processed sequentially in Envoy, so if this "global" rule is not the first one in the rules, it won't take effect.